Non-keyed implementation

mksunny1 opened the pull request:
- added non-keyed implementation for Vanillajs-3
- small code enhancements
- semantic bug fix (in remove) for the keyed implementation.

krausest commented:
I plan to publish results tomorrow.

krausest commented:
Left new keyed results, right old keyed results. Nice!
![Screenshot 2024-05-16 at 19 51 44](https://github.com/krausest/js-framework-benchmark/assets/3374055/6536995d-d747-4054-9ad1-af874bbcf76b)

mksunny1 commented:
So sorry I introduced a bigger semantic bug when fixing the first one. I have now corrected it. Additionally, I updated the `append` function to clone the item template retrieved from DOM before using it. Without cloning, it will grow to a count of 100 when adding 10K rows hogging memory and affecting overall performance (I think)...
